src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -240,24 +299,201 @@ private String generateFrameDescription(WindowAggregationConfig config) {
         return ".";
       case ROW:
         return String.format("over a windows frame consisting of %s preceding and %s following rows.",
-                             isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
-                             isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
+          isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
+          isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
       case RANGE:
         return String.format("over a windows frame consisting of %s preceding and %s following range.",
-                             isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
-                             isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
+          isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
+          isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
     }
 
     return "";
   }
 
   @Override
-  public JavaRDD<StructuredRecord> transform(
-    SparkExecutionPluginContext sparkExecutionPluginContext, JavaRDD<StructuredRecord> javaRDD) throws Exception {
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext sparkExecutionPluginContext,
+                                             JavaRDD<StructuredRecord> javaRDD) throws Exception {
     Schema inputSchema = sparkExecutionPluginContext.getInputSchema();
     if (inputSchema == null) {
       throw new Exception("Input schema is null. Input schema can not be null at this stage");
     }
     return WindowsAggregationUtil.transform(sparkExecutionPluginContext, javaRDD, config, inputSchema, outputSchema);
   }
+
+  @Override
+  public boolean canUseEngine(Engine engine) {
+    Optional<ExpressionFactory<String>> expressionFactory = engine.
+      getExpressionFactory(StringExpressionFactoryType.SQL);
```

Comment:
You also need the BigQuery capability here, as your aggregations use BigQuery syntax.

Reply:
You can also check if the engine exposes the BigQuery capability before responding true.

Reply:
Done